src/hooks/prisma.ts

```diff
@@ -10,15 +10,39 @@ import { FunctionInfo } from "../registry";
 import config from "../config";
 
 export default function prismaHook(mod: typeof prisma, id?: string) {
+  assert(mod.PrismaClient != null);
+  // (1) Prisma Queries: We proxy prismaClient._request method in order to record
+  // prisma queries (not sqls) as appmap function call events.
+  // (2) SQL Queries: We have to change config parameters (logLevel, logQueries)
+  // and register a prismaClient.$on("query") handler to record sql queries.
+  // We have to do it by proxying mod.PrismaClient here, since it turned out that
+  // it's too late to do it inside the first invocation of the _request method,
+  // because $on is (becomes?) undefined in extended prisma clients.
+  // https://www.prisma.io/docs/orm/reference/prisma-client-reference#remarks-37
+
+  // Normally, we "Cannot assign to 'PrismaClient' because it is a read-only property."
+  // Fortunately with this hack it can be done.
+  const modHacked = mod as unknown as { PrismaClient: unknown };
+  modHacked.PrismaClient = new Proxy(mod.PrismaClient, {
+    construct(target, argArray, newTarget): object {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-argument
+      const result = Reflect.construct(target, argArray, newTarget);
+      if (!hookAttached) {
+        hookAttached = true;
```

Comment:
I didn't notice it before, but now I can see `hookAttached` is a global. I know it's an edge case, but it will break if there are multiple prisma clients (as can be configured with multiple `prisma:` entries in the config file). How about using a `WeakSet` instead to keep track of patched clients?